miniapps/shifted/lsf_integral.cpp

```diff
@@ -183,58 +183,39 @@ int main(int argc, char *argv[])
 
 #ifdef MFEM_USE_ALGOIM
    real_t area=0.0;
-   DenseMatrix bmat; // gradients of the shape functions in isoparametric space
-   DenseMatrix pmat; // gradients of the shape functions in physical space
-   Vector inormal; // normal to the level set in isoparametric space
-   Vector tnormal; // normal to the level set in physical space
-   Vector lsfun; // level set function restricted to an element
-   DofTransformation *doftrans;
-   Array<int> vdofs;
+
+   AlgoimIntegrationRules* air=new AlgoimIntegrationRules(aorder,*ls_coeff,order);
+
+   IntegrationRule eir;
+   Vector sweights;
+
    for (int i=0; i<fespace.GetNE(); i++)
    {
-      const FiniteElement* el=fespace.GetFE(i);
-
       // get the element transformation
       trans = fespace.GetElementTransformation(i);
 
-      // extract the element vector from the level-set
-      doftrans = fespace.GetElementVDofs(i,vdofs);
-      x.GetSubVector(vdofs, lsfun);
-
-      // construct Algoim integration object
-      AlgoimIntegrationRule air(aorder, *el, *trans, lsfun);
-
-      // compute the volume contribution from the element
-      ir = air.GetVolumeIntegrationRule();
-      for (int j = 0; j < ir->GetNPoints(); j++)
+      air->GetVolumeIntegrationRule(*trans,eir);
```

Comment:
To aid future user, we could skip this for elements that are not cut by checking the level-set min and max.

Reply:
It returns standard integration rule if the element is not cut - the level set is computed in the construction of the integration rule. If we modify the code it will require markings - and these will be introduced in an upcoming cutfem miniapp.  For demonstrating the capability the current interface is more clean.